.Dislocation Motion.tmp.nlogo

```diff
@@ -275,104 +283,113 @@ to calculate-fl-positions
   ]
 end
 
-to identify-force-atoms
+to identify-force-atoms ; (find the atoms closest to the force line that will be the ones receiving the external force)
   (ifelse force-mode = "Shear" [
-    ask atoms [ set ex-force-applied?  0 ]
-    ask min-n-of num-forced-atoms atoms with [ ycor >= median-ycor ] [ distancexy upper-left-fl ycor] [
-      set ex-force-applied?  1
+    ask atoms [ set ex-force-applied?  False ]
+    let forced-atoms atoms with [ ycor >= median-ycor and (distancexy upper-left-fl ycor) <= 1]
+    set num-forced-atoms count forced-atoms
+    ask forced-atoms [
+      set ex-force-applied?  True
     ]
     ]
     force-mode = "Compression" [
-      ask atoms [ set ex-force-applied?  0 ]
-      ask min-n-of num-forced-atoms atoms [ distancexy left-fl ycor] [
-        set ex-force-applied?  1
+      ask atoms [ set ex-force-applied?  False ]
+      let forced-atoms atoms with [ (distancexy left-fl ycor) <= 1]
+      set num-forced-atoms count forced-atoms
+      ask forced-atoms [
+        set ex-force-applied?  True
     ]
   ]) ; for tension, the same atoms in the left shoulder of the sample always receive the force
 end
 
 
 to check-eq-adj-force ; (check equilibrium, adjust force)
```

Comment:
Let's just call this adjust-force. If it isn't yet close to equilibrium, the adjustment is zero. I think it is more descriptive/clearer name.